packages/flutter/lib/src/widgets/magnifier.dart

```diff
@@ -477,86 +546,53 @@ class RawMagnifier extends StatelessWidget {
             ),
           ),
         ),
-        // Because `BackdropFilter` will filter any widgets before it, we should
-        // apply the style after (i.e. in a younger sibling) to avoid the magnifier
+        // Because `BackdropFilter` will filter any widgets before it, we apply
+        // these styles after (i.e. in a younger sibling) to avoid the magnifier
         // from seeing its own styling.
-        Opacity(
-          opacity: decoration.opacity,
-          child: _MagnifierStyle(
-            decoration,
-            size: size,
+        IgnorePointer(
+          child: Opacity(
+            opacity: decoration.opacity,
+            child: ClipPath(
+              clipBehavior: clipBehavior,
+              clipper: _NegativeClip(shape: decoration.shape),
+              child: DecoratedBox(
+                decoration: ShapeDecoration(
+                  shape: decoration.shape,
+                  shadows: decoration.shadows,
+                ),
+                child: SizedBox.fromSize(
+                  size: size,
+                ),
+              ),
+            ),
           ),
-        )
-      ],
-    );
-  }
-}
-
-class _MagnifierStyle extends StatelessWidget {
-  const _MagnifierStyle(this.decoration, {required this.size});
-
-  final MagnifierDecoration decoration;
-  final Size size;
-
-  @override
-  Widget build(BuildContext context) {
-    double largestShadow = 0;
-    for (final BoxShadow shadow in decoration.shadows ?? <BoxShadow>[]) {
-      largestShadow = math.max(
-          largestShadow,
-          (shadow.blurRadius + shadow.spreadRadius) +
-              math.max(shadow.offset.dy.abs(), shadow.offset.dx.abs()));
-    }
-
-    return ClipPath(
-      clipBehavior: Clip.hardEdge,
-      clipper: _DonutClip(
-        shape: decoration.shape,
-        spreadRadius: largestShadow,
-      ),
-      child: DecoratedBox(
-        decoration: decoration,
-        child: SizedBox.fromSize(
-          size: size,
         ),
-      ),
+      ],
     );
   }
 }
 
-/// A `clipPath` that looks like a donut if you were to fill its area.
-///
-/// This is necessary because the shadow must be added after the magnifier is drawn,
-/// so that the shadow does not end up in the magnifier. Without this clip, the magnifier would be
-/// entirely covered by the shadow.
-///
-/// The negative space of the donut is clipped out (the donut hole, outside the donut).
-/// The donut hole is cut out exactly like the shape of the magnifier.
-class _DonutClip extends CustomClipper<Path> {
-  _DonutClip({required this.shape, required this.spreadRadius});
+// A clip that renders everything except the inside of a shape.
```

Comment:
I'll miss "donut" but that does sound better and more generic :doughnut:

Reply:
well also it stopped looking like a donut once i made it use an infinite rectangle instead of just growing the path to just enough to cover everything. :-)